lib/SILOptimizer/Utils/Devirtualize.cpp

```diff
@@ -91,7 +91,8 @@ static bool isEffectivelyFinalMethod(FullApplySite applySite, CanType classType,
 
   auto *cmi = cast<MethodInst>(applySite.getCallee());
 
-  if (!calleesAreStaticallyKnowable(applySite.getModule(), cmi->getMember()))
+  if (!calleesAreStaticallyKnowable(applySite.getModule(), cmi->getMember()) &&
+      (!cd || cd->getFormalAccess() != AccessLevel::Package))
```

Comment:
I think getEffectiveAccess() check is more accurate in SIL optimizer.  Also don't we want the same for `public` access? It also can't be subclassed or overriden outside of defining module.

Reply:
Why is `package` access special (i.e different to `internal` access)?

If the code following this check depends on `module.isWholeModule()` for `internal` and `public` access (see implementation `calleesAreStaticallyKnowable`) in one of the calls (which I assume might be the case), then why is this not the case for package access level?